### PR TITLE
Clean up Detection and Segmentation code

### PR DESCRIPTION
**Pull request by fyu**
- [x] Remove redundant detection code
- [x] Move segmentation to a separate file
- [x] Unit tests for semantic segmentation

### CODE REVIEW

File: bdd100k/eval/detect_test.py [hunk 1]
```diff
@@ -11,24 +11,22 @@ class TestBDD100KDetectEval(unittest.TestCase):
...
-            "AP_small": 0.18365219335000335,
-            "AP_medium": 0.4841288233364759,
-            "AP_large": 0.6320439295297166,
-            "AR_max_1": 0.21599709015331267,
-            "AR_max_10": 0.3531322234358392,
-            "AR_max_100": 0.3793465642222695,
-            "AR_small": 0.22512489216925038,
-            "AR_medium": 0.553898980511434,
-            "AR_large": 0.6504448777029422,
+            "AP": 0.3402939266840102,
```
Comment (fyu) [inline bdd100k/eval/detect_test.py:18]: @XiaLiPKU Why is the testing result updated?

File: bdd100k/eval/seg_test.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import numpy as np
+
+from .seg import evaluate_segmentation, fast_hist
+
+
+class TestIoU(unittest.TestCase):
+    """Test cases for the segmentation IoU computation."""
+
+    def test_fast_hist(self) -> None:
```
Comment (fyu) [inline bdd100k/eval/seg_test.py:13]: It is more concise to use pytest syntax here

File: bdd100k/eval/detect_test.py (hunk 1 above)
Comment (XiaLiPKU) [reply bdd100k/eval/detect_test.py:18]: Previously, we use coco format gt for evaluation. These coco format GTs, are generated with off-by-one errors.
Now, we use the original BDD format test file. So there're obvious differences.

File: bdd100k/eval/seg_test.py (hunk 2 above)
Comment (XiaLiPKU) [reply bdd100k/eval/seg_test.py:13]: Can I see some examples? I'm not familiar with pytest syntax

Comment (fyu) [reply bdd100k/eval/seg_test.py:13]: Here are some https://github.com/scalabel/scalabel/blob/master/scalabel/label/io_test.py 🙂

**Pull request merged**